### PR TITLE
Don't iterate the whole asset graph to generate asset condition evaluation records

### PR DESCRIPTION
**Pull request by gibsondan**
BK

## Changelog

> Insert changelog entry or delete this section.

### CODE REVIEW

File: python_modules/dagster-graphql/dagster_graphql/implementation/fetch_asset_condition_evaluations.py [hunk 1]
```diff
@@ -45,12 +44,11 @@ def _get_graphene_records_from_evaluations(
 
     partitions_defs = {}
 
-    nodes = get_asset_nodes_by_asset_key(graphene_info)
     for asset_key in asset_keys:
-        asset_node = nodes.get(asset_key)
+        remote_node = graphene_info.context.asset_graph.get(asset_key)
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql/implementation/fetch_asset_condition_evaluations.py:48]: `get` throws on miss and this existing code handles missing keys so need a `has` check here first

**Comment by gibsondan** [comment]
hm I could certainly match the existing logic but @OwenKephart do we still even want to return records in this case? It seems weird to pass in a None partitions def but still return records, do we handle that well?

**Comment by gibsondan** [comment]
ah wait we didn't even need to be fetching assets at all!

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_condition_evaluations.py [hunk 2]
```diff
@@ -272,7 +271,6 @@ class Meta:
     def __init__(
         self,
         record: AutoMaterializeAssetEvaluationRecord,
-        partitions_def: Optional[PartitionsDefinition],
```
Comment (gibsondan) [inline python_modules/dagster-graphql/dagster_graphql/schema/asset_condition_evaluations.py:275]: unused!

**Review by OwenKephart — APPROVED** [review]
Let's make sure to get to the bottom of who added that unused parameter 🤔

**Pull request merged**